pkg/wallet/account/account.go

```diff
@@ -0,0 +1,265 @@
+package account
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"fmt"
+
+	"github.com/awnumar/memguard"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/station-massa-wallet/pkg/crypto"
+	"github.com/massalabs/station-massa-wallet/pkg/types"
+	"github.com/massalabs/station-massa-wallet/pkg/types/object"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	AccountLastVersion = 1
+)
+
+type Account struct {
+	Version      *uint8                    `yaml:"Version"`
+	Nickname     string                    `yaml:"Nickname,omitempty"`
+	Address      types.Address             `yaml:"Address,omitempty"`
+	Salt         [16]byte                  `yaml:"Salt,flow"`
+	Nonce        [12]byte                  `yaml:"Nonce,flow"`
+	CipheredData types.EncryptedPrivateKey `yaml:"CipheredData,flow"`
+	PublicKey    types.PublicKey           `yaml:"PublicKey,flow"`
+}
+
+func New(version *uint8, nickname string, address types.Address, salt [16]byte, nonce [12]byte, cipheredData types.EncryptedPrivateKey, publicKey types.PublicKey) (*Account, error) {
```

Comment:
Why address, types.EncryptedPrivateKey, publicKey are not pass by refenrece instead of duplicating them ?

Furthermore cipheredData is obscure, you should use encryptedPrivateKey as you are using address or publickey.